third_party/move/move-compiler-v2/src/pipeline/reference_safety_processor.rs

```diff
@@ -1012,7 +1012,7 @@ impl<'env, 'state> LifetimeAnalysisStep<'env, 'state> {
     fn check_borrow_safety(&mut self, temps_vec: &[TempIndex]) {
         // First check direct duplicates
         for (i, temp) in temps_vec.iter().enumerate() {
-            if temps_vec[i + 1..].contains(temp) {
+            if self.ty(*temp).is_mutable_reference() && temps_vec[i + 1..].contains(temp) {
```

Comment:
*** This also is a pretty big deal for borrow analysis.

Reply:
Yes, it was a serious bug. The one-and-only the prover integration found so far.